examples/end_to_end_distance.py

```diff
@@ -0,0 +1,97 @@
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Calculate and plot end-to-end distance of neurites.'''
+
+from neurom.analysis import morphmath
+from neurom import ezy
+from neurom.core import tree
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def segment_vector(seg):
+    '''Calculate and return segment vector.'''
+    return np.subtract(seg[1][0:3], seg[0][0:3])
+
+
+def neurite_end_to_end_distance(neurite):
+    '''Calculate and return end-to-end-distance of a given neurite.'''
+    trunk = neurite.value
+    return max(morphmath.point_dist(l, trunk) for l in tree.val_iter(tree.ileaf(neurite)))
+
+
+def mean_end_to_end_dist(neurites):
+    '''Calculate mean end to end distance for set of neurites.'''
+    return np.mean([neurite_end_to_end_distance(n) for n in neurites])
+
+
+def make_end_to_end_distance_plot(nb_segments, end_to_end_distance, neurite_type):
+    '''Plot end-to-end distance vs number of segments'''
+    plt.figure()
+    plt.plot(nb_segments, end_to_end_distance)
+    plt.title(neurite_type)
+    plt.xlabel('Number of segments')
+    plt.ylabel('End-to-end distance')
+    plt.show()
+
+
+def calculate_and_plot_end_to_end_distance(neurite):
+    '''Calculate and plot the end-to-end distance vs the number of segments for
+    an increasingly larger part of a given neurite.'''
+    n = sum(1 for _ in tree.isegment(neurite))
+    end_to_end_distance = np.zeros(n)
+    index = 0
+    max_so_far = 0
+    for segment in tree.val_iter(tree.isegment(neurite)):
+        max_so_far = max(max_so_far, morphmath.point_dist(segment[1], neurite.value))
+        end_to_end_distance[index] = max_so_far
+        index += 1
```

Comment:
I'd like to make this as compact as the one on line 46, but I don't know how I can select an increasingly larger part of a neurite...